expression/util.go

```diff
@@ -449,3 +450,19 @@ func (s *exprStack) push(expr Expression) {
 func (s *exprStack) len() int {
 	return len(s.stack)
 }
+
+// ColumnSliceIntersect intersects two column slice.
+// You need to make sure that at least each element in s2 is unique.
+func ColumnSliceIntersect(s1, s2 []*Column) []*Column {
+	intSet := map[int]struct{}{}
+	for _, col := range s1 {
+		intSet[col.UniqueID] = struct{}{}
+	}
```

Comment:
UniqueID is changed to int64 in https://github.com/pingcap/tidb/pull/7385, should modify the map definition here.